### ISSUE TITLE
TypeError: object() takes no parameters in shadow.py

### ISSUE THREAD

**Issue by Eshan-Agarwal** [cython/cython#3401]
I am using tensorflow and it uses apache beam, when I run code I got this error
```
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\site-packages\Cython\Shadow.py", line 165, in cast
    return type(*args)
TypeError: object() takes no parameters
```

**Comment by Eshan-Agarwal** [comment]
Error is in `shadow.py` file in

> def cast(type, *args, **kwargs):
    kwargs.pop('typecheck', None)
    assert not kwargs
    if hasattr(type, '__call__'):
        return type(*args)
    else:
        return args[0]

**Comment by scoder** [comment]
> when I run code

What kind of code? Would you have an example?

**Comment by scoder** [comment]
… or a longer stack track, at least? Anything that would show _what_ is going wrong?

**Comment by Eshan-Agarwal** [comment]
@scoder Thanks for checking out but I found this PR which solves this issue #3244

**Comment by Eshan-Agarwal** [comment]
@scoder  here is code that is used for downloading data `wikipedia` 
`python -m tensorflow_datasets.scripts.download_and_prepare --register_checksums --datasets=wikipedia/20200301.bn`

and this is full error : 

> Traceback (most recent call last):
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\eshan\Desktop\Tensoflow\Tfds\datasets\tensorflow_datasets\scripts\download_and_prepare.py", line 219, in <module>
    app.run(main)
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\site-packages\absl\app.py", line 299, in run
    _run_main(main, args)
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\site-packages\absl\app.py", line 250, in _run_main
    sys.exit(main(argv))
  File "C:\Users\eshan\Desktop\Tensoflow\Tfds\datasets\tensorflow_datasets\scripts\download_and_prepare.py", line 214, in main
    download_and_prepare(builder)
  File "C:\Users\eshan\Desktop\Tensoflow\Tfds\datasets\tensorflow_datasets\scripts\download_and_prepare.py", line 144, in download_and_prepare
    download_config=dl_config,
  File "C:\Users\eshan\Desktop\Tensoflow\Tfds\datasets\tensorflow_datasets\core\api_utils.py", line 52, in disallow_positional_args_dec
    return fn(*args, **kwargs)
  File "C:\Users\eshan\Desktop\Tensoflow\Tfds\datasets\tensorflow_datasets\core\dataset_builder.py", line 340, in download_and_prepare
    download_config=download_config)
  File "C:\Users\eshan\Desktop\Tensoflow\Tfds\datasets\tensorflow_datasets\core\dataset_builder.py", line 1193, in _download_and_prepare
    pipeline=pipeline,
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\site-packages\apache_beam\pipeline.py", line 481, in __exit__
    self.run().wait_until_finish()
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\site-packages\apache_beam\pipeline.py", line 461, in run
    self._options).run(False)
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\site-packages\apache_beam\pipeline.py", line 474, in run
    return self.runner.run_pipeline(self, self._options)
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\site-packages\apache_beam\runners\direct\direct_runner.py", line 182, in run_pipeline
    return runner.run_pipeline(pipeline, options)
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\site-packages\apache_beam\runners\portability\fn_api_runner.py", line 486, in run_pipeline
    default_environment=self._default_environment))
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\site-packages\apache_beam\runners\portability\fn_api_runner.py", line 494, in run_via_runner_api
    return self.run_stages(stage_context, stages)
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\site-packages\apache_beam\runners\portability\fn_api_runner.py", line 583, in run_stages
    stage_context.safe_coders)
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\site-packages\apache_beam\runners\portability\fn_api_runner.py", line 904, in _run_stage
    result, splits = bundle_manager.process_bundle(data_input, data_output)
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\site-packages\apache_beam\runners\portability\fn_api_runner.py", line 2105, in process_bundle
    for result, split_result in executor.map(execute, part_inputs):
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\concurrent\futures\_base.py", line 586, in result_iterator
    yield fs.pop().result()
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\concurrent\futures\_base.py", line 432, in result
    return self.__get_result()
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\concurrent\futures\_base.py", line 384, in __get_result
    raise self._exception
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\site-packages\apache_beam\utils\thread_pool_executor.py", line 44, in run
    self._future.set_result(self._fn(*self._fn_args, **self._fn_kwargs))
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\site-packages\apache_beam\runners\portability\fn_api_runner.py", line 2102, in execute
    return bundle_manager.process_bundle(part_map, expected_outputs)
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\site-packages\apache_beam\runners\portability\fn_api_runner.py", line 2025, in process_bundle
    result_future = self._worker_handler.control_conn.push(process_bundle_req)
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\site-packages\apache_beam\runners\portability\fn_api_runner.py", line 1358, in push
    response = self.worker.do_instruction(request)
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\site-packages\apache_beam\runners\worker\sdk_worker.py", line 352, in do_instruction
    request.instruction_id)
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\site-packages\apache_beam\runners\worker\sdk_worker.py", line 386, in process_bundle
    bundle_processor.process_bundle(instruction_id))
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\site-packages\apache_beam\runners\worker\bundle_processor.py", line 812, in process_bundle
    data.transform_id].process_encoded(data.data)
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\site-packages\apache_beam\runners\worker\bundle_processor.py", line 205, in process_encoded
    self.output(decoded_value)
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\site-packages\apache_beam\runners\worker\operations.py", line 304, in output
    cython.cast(Receiver, self.receivers[output_index]).receive(windowed_value)
  File "C:\Users\eshan\Anaconda3\envs\keras-gpu\lib\site-packages\Cython\Shadow.py", line 165, in cast
    return type(*args)
TypeError: object() takes no parameters

**Comment by scoder** [comment]
I pushed 3149eb8367e69f1558c73b0ebc266bdfba8abc0d, just in case.
It looks like they intended for this code to be compiled, though, but it isn't. It runs in Python instead of native code. Maybe something went wrong on your side when installing Apache Beam? Or maybe they did something wrong with the release on their side?

**Comment by Eshan-Agarwal** [comment]
so should i correct it according to `3149eb8` or reinstall `Apache Beam`

**Comment by black-square** [comment]
@scoder what do you think about accepting PR #3244? It fixes this and another more fundamental problem. I expressed everything in the additional tests

**Comment by scoder** [comment]
Resolved in #3244.
